### PR TITLE
[Keyboard] Convert most of sinc to data driven info.json

### PR DESCRIPTION
**Pull request by SethBarberee**
## Description

Converted most of sinc to data driven config. I have a Rev1 so was able to test locally with my keymap. Not everything was converted because some stuff isn't 1:1 yet such as `RGBLED_MAP` and `RGBLIGHT_DEFAULT_MODE`

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/keebio/sinc/rev2/config.h [hunk 1]
```diff
@@ -21,37 +21,17 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define MATRIX_ROWS 12
 #define MATRIX_COLS 9
```
Comment (fauxpark) [inline keyboards/keebio/sinc/rev2/config.h:22]:
```suggestion
```
CLI should be able to figure this out.

File: keyboards/keebio/sinc/rev2/config.h [hunk 2]
```diff
@@ -21,37 +21,17 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
-#define ENCODERS_PAD_A { F5 }
-#define ENCODERS_PAD_B { F6 }
-#define ENCODERS_PAD_A_RIGHT { D6 }
-#define ENCODERS_PAD_B_RIGHT { D4 }
 
-/* Set 0 if debouncing isn't needed */
-#define DEBOUNCE 5
-
-#define BACKLIGHT_PIN B5
 #define CAPS_LOCK_LED_PIN B6
```
Comment (fauxpark) [inline keyboards/keebio/sinc/rev2/config.h:27]: This can be moved too.

File: keyboards/keebio/sinc/rev1/config.h [hunk 3]
```diff
@@ -23,37 +23,17 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
-#define ENCODERS_PAD_A { F5 }
-#define ENCODERS_PAD_B { F6 }
-#define ENCODERS_PAD_A_RIGHT { D6 }
-#define ENCODERS_PAD_B_RIGHT { D4 }
 
-/* Set 0 if debouncing isn't needed */
-#define DEBOUNCE 5
-
-#define BACKLIGHT_PIN B5
 #define CAPS_LOCK_LED_PIN B6
```
Comment (fauxpark) [inline keyboards/keebio/sinc/rev1/config.h:29]: As above.

File: keyboards/keebio/sinc/rev1/config.h [hunk 4]
```diff
@@ -23,37 +23,17 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define MATRIX_ROWS 12
 #define MATRIX_COLS 9
```
Comment (fauxpark) [inline keyboards/keebio/sinc/rev1/config.h:24]: As above.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keebio/sinc/rev2/config.h (hunk 2 above)
Comment (SethBarberee) [reply keyboards/keebio/sinc/rev2/config.h]: Fixed

File: keyboards/keebio/sinc/rev2/config.h (hunk 1 above)
Comment (SethBarberee) [reply keyboards/keebio/sinc/rev2/config.h]: Fixed

**Review by fauxpark — APPROVED** [review]

**Pull request merged**